src/agent0/core/hyperdrive/policies/simple_lp.py

```diff
@@ -0,0 +1,172 @@
+        Each time `action` is called is a step.
+        """
+
+    def __init__(
+        self,
+        policy_config: Config,
+    ):
+        """Initialize the bot.
+
+        Arguments
+        ---------
+        policy_config: Config
+            The custom arguments for this policy
+        """
+        super().__init__(policy_config)
+        assert policy_config.pnl_target > FixedPoint(0), "PNL target must be greater than zero."
+        self.policy_config = policy_config
+        self.pnl_history: list[tuple[FixedPoint, FixedPoint]] = []
+        self.total_base_spent: FixedPoint = FixedPoint(0)
+
+    def action(
+        self, interface: HyperdriveReadInterface, wallet: HyperdriveWallet
+    ) -> tuple[list[Trade[HyperdriveMarketAction]], bool]:
+        """Specify actions.
+
+        Arguments
+        ---------
+        interface: HyperdriveReadInterface
+            Interface for the market on which this agent will be executing trades (MarketActions).
+        wallet: HyperdriveWallet
+            The agent's wallet.
+
+        Returns
+        -------
+        tuple[list[MarketAction], bool]
+            A tuple where the first element is a list of actions,
+            and the second element defines if the agent is done trading.
+        """
+        # Get the current state of the pool & the bot's position
+        current_block = interface.get_current_block()
+        pool_state = interface.get_hyperdrive_state(current_block)
+        current_holding = wallet.lp_tokens * pool_state.pool_info.lp_share_price
+
+        # Need to be in the game to play it
+        if self.policy_config.minimum_liquidity_tokens > current_holding:
+            trade_amount = self.policy_config.minimum_liquidity_tokens - current_holding
+            return [add_liquidity_trade(trade_amount)], False
+
+        # Get current PNL
+        pnl = current_holding - self.total_base_spent
+        self.pnl_history.append((FixedPoint(interface.get_block_timestamp(current_block)), pnl))
+        # Prune history
+        if FixedPoint(len(self.pnl_history)) > self.policy_config.lookback_length:
+            self.pnl_history = self.pnl_history[-self.policy_config.lookback_length :]
+        # PNL slope is dy/dx, where y is PNL and x is time.
+        # We compute an average of splines instead of a global average
+        # to reduce emphasis on the first and last time-points.
+        avg_pnl_slope = 0
+        if len(self.pnl_history) > 2:
+            for i in range(len(self.pnl_history) - 1):
+                dx = self.pnl_history[i + 1][0] - self.pnl_history[i][0]
+                dy = self.pnl_history[i + 1][1] - self.pnl_history[i][1]
+                avg_pnl_slope += dy / dx
+            avg_pnl_slope /= len(self.pnl_history) - 1
+
+        # Make trades based on pnl_target and avg_pnl_slope
+        action_list = []
+
+        # I'm doing great, keep putting money in
+        if pnl > self.policy_config.pnl_target and avg_pnl_slope > 0:
```

Comment:
pnl is a stock and slope is a flow. I think you want the pnl target on the flow measure.

any stock-based measure could become useless as your total wealth moves far away from that point.

Reply:
to keep `SimpleLP` as simple as possible, I'd simplify the logic to just the flow measure, and accept they'll always add or remove. or explicitly define a do-nothing range, which in this case would be in units of base/second I think. or do my original suggestion to make your target liquidity proportional to your rate of return relative to a high target and your budget: `target_liquidity = budget * max(0,min(rate_of_return / 10%,1))` which allows for varying trade sizes and faster convergence.

Reply:
hmm this is my first time hearing "stock" vs "flow", but I do think the direction we _were_ going with this was the slope as the metric. However, I had a hard time thinking about what units made sense and so I switched to the "stock" value. I think it's worth considering this metric further, but if agree that this is good enough then I'd like to avoid reworking it again in favor of getting an mvp out the door.

Reply:
sounds good, fine with not reworking